File: src/main/java/ru/ioleksiv/telegram/bot/core/api/model/method/update/GetWebhookInfo.java
package ru.ioleksiv.telegram.bot.core.api.model.method.update;

import ru.ioleksiv.telegram.bot.core.api.model.objects.WebhookInfo;
import ru.ioleksiv.telegram.bot.core.api.model.NetworkError;
import ru.ioleksiv.telegram.bot.core.model.method.RunnableAction;
import ru.ioleksiv.telegram.bot.core.model.responses.CommonResponse;
import ru.ioleksiv.telegram.bot.core.model.responses.ResponseCollection;
import ru.ioleksiv.telegram.bot.core.controller.network.Networker;

/**
 * @see <a href="https://core.telegram.org/bots/api#getwebhookinfo">getWebhookInfo</a>
 */
public class GetWebhookInfo extends RunnableAction<WebhookInfo> {
    private static final String METHOD = "getWebhookInfo";

    public GetWebhookInfo(Networker networker) {
        super(METHOD, networker);
    }

    @Override
    protected Class<? extends CommonResponse<WebhookInfo>> getResultWrapperClass() {
        return ResponseCollection.WebhookInfoResponse.class;
    }

    @Override
    public GetWebhookInfo setNetworkErrorListener(NetworkError onNetworkError) {
        pSetNetworkErrorListener(onNetworkError);
        return this;
    }
}

File: server-api/src/main/kotlin/io/github/oleksivio/tl/kbot/server/api/method/update/GetWebhookInfo.kt
package io.github.oleksivio.tl.kbot.server.api.method.update

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.oleksivio.tl.kbot.server.api.model.ApiDict
import io.github.oleksivio.tl.kbot.server.api.model.WebhookInfoResponse
import io.github.oleksivio.tl.kbot.server.api.model.method.Action
import io.github.oleksivio.tl.kbot.server.api.objects.WebhookInfo

/**
 * @see [getWebhookInfo](https://core.telegram.org/bots/api/#getwebhookinfo)
 */
class GetWebhookInfo : Action<WebhookInfo>() {

    @JsonProperty(ApiDict.METHOD_KEY)
    override val method: String = "getWebhookInfo"

    @JsonIgnore
    override val resultWrapperClass = WebhookInfoResponse::class
}
